src/setup.ts

```diff
@@ -5,8 +5,11 @@ import { createElement } from 'react';
 
 import useTheme from './theme';
 
+import type { StyleProps } from './types/quarkProps';
+
 const stylePropPrefix = '$';
-export const validateProp = (key: string) => key.startsWith(stylePropPrefix);
+export const validateProp = (string: string): string is keyof StyleProps =>
+  typeof string === 'string' && string.startsWith(stylePropPrefix);
```

Comment:
```suggestion
export const validateProp = (string: string): string is keyof StyleProps => string.startsWith(stylePropPrefix);
```
Either we should change the type for the input or remove the type check. I don't think it should be both. Having both says that we have the wrong type for the input.

Reply:
Not sure I understand what you're asking. I remove the type check already.

Reply:
What is the point of `typeof string === 'string'` if the function ONLY allows `strings`? Shouldn't string always be of type `string`? Thus making that check redundant. It's either that, or for some reason something OTHER than a string is getting passed into his function. That then would make the prop type `validateProp = (string: string)` wrong. Right?